cs_tools/thoughtspot.py

```diff
@@ -35,7 +35,7 @@ def __init__(self, config):
         self.metadata = MetadataMiddleware(self)
         self.pinboard = PinboardMiddleware(self)
         self.answer = AnswerMiddleware(self)
-        # self.connection
+        self.connection = ConnectionMiddleware(self)
```

Comment:
something to think about as you expand this tool over the next few months, we can have a TML middleware that does a lot of the common error handling. it'd might feel like an extension of Bryant's TML library, but essentially if you're seeing common patterns that we can simplify with a middleware, let's do it.

Reply:
This is a good idea.  I think it might be good in v2, to revisit some of the architecture and extract what's common.

Reply:
Or rather, extract what is reusable logic.